Update layout_stride_fix with LWG feedback (#331)
ORNL/cpp-proposals-pub: layout_stride_fix/layout_stride_fix.md

# Revision History

# Revision 1:

> - LWG just wants the default constructor to always do what was proposed for all-static extents case

Reviewer: Optionally:

```suggestion
- LWG wants the default constructor to behave the same regardless of whether all the extents are static: that is, it should fill in the strides in the same way as `layout_right::mapping` would.
```

We believe that it is preferable to preserve default constructibility of `layout_stride` for all specializations of `layout_stride` to simplify a number of generic programming cases -- specifically, if layout policies are used directly in higher-level data structures for which the user wants to enable default constructibility.

> Based on feedback in LWG we leave out the defaulted constructor for cases with dynamic extents and just always default construct layout_stride as if it were a layout_right with the same default constructed extents.

Reviewer: ```suggestion
Based on feedback in LWG we leave out the defaulted constructor for cases with dynamic extents and just always default construct `layout_stride::mapping` as if it were a `layout_right::mapping` with the same default constructed extents.
```

> Based on feedback in LWG we leave out the defaulted constructor for cases with dynamic extents and just always default construct layout_stride as if it were a layout_right with the same default constructed extents.

Reviewer 2: ```suggestion
Based on feedback in LWG, we leave out the defaulted constructor for cases with dynamic extents and just always default construct `layout_stride` as if it were a `layout_right` with the same default constructed extents.
```

With:
```c++
    // [mdspan.layout.stride.cons], constructors
    constexpr mapping() noexcept; 
    constexpr mapping(const mapping&) noexcept = default;
```

Add at the beginning of subsection 24.7.3.4.7.3 [mdspan.layout.stride.cons] insert:

```c++
    constexpr mapping() noexcept;
```

> *Effects:* Direct-non-list-initializes _`extents_`_ with `extents_type()` , and for all $d$ in the range $[$`0, `_`rank_`_$)$, 

Reviewer: ```
*Preconditions*: `layout_right::mapping<extents_type>{}.required_­span_­size()` is representable as a value of type `index_­type` ([[basic.fundamental]](https://eel.is/c++draft/basic.fundamental)).
```
